lib/driver.js

```diff
@@ -600,14 +612,16 @@ class XCUITestDriver extends BaseDriver {
     await this.installAUT();
 
     // if we only have bundle identifier and no app, fail if it is not already installed
-    if (
-      !this.opts.app &&
-      this.opts.bundleId &&
-      !this.isSafari() &&
-      // @ts-expect-error - do not assign arbitrary properties to `this.opts`
-      !(await this.opts.device.isAppInstalled(this.opts.bundleId))
-    ) {
-      this.log.errorAndThrow(`App with bundle identifier '${this.opts.bundleId}' unknown`);
+    if (!this.skipDeviceCheck()) {
+      if (
```

Comment:
This condition could be combined. Nested ifs are hard to read

Reply:
yea, let me modify this pr a bit later